lab2.js

```diff
@@ -86,33 +104,71 @@ var hello = {
 // speak, and method (that you'll place on the prototype) called
 // sayHello.
 
-function SentientBeing () {
-  // TODO: specify a home planet and a language
+
+// TODO: specify a home planet and a language
   // you'll need to add parameters to this constructor
+
+function SentientBeing (homePlanet, language) {
+  this.homePlanet = homePlanet;
+  this.language = language;
 }
 
 // sb is a SentientBeing object
-function sayHello (sb) {
+SentientBeing.prototype.sayHello = function sayHello (sb) {
     // TODO: say hello prints out (console.log's) hello in the
     // language of the speaker, but returns it in the language
     // of the listener (the sb parameter above).
     // use the 'hello' object at the beginning of this exercise
     // to do the translating
+   if (sb.language == 'romulan') {
+    console.log(hello[this.language]);
+    return (hello.romulan);
+
+  }
 
-    //TODO: put this on the SentientBeing prototype
+  if (sb.language == 'klingon') {
+    console.log(hello[this.language]);
+    return (hello.klingon);
   }
 
+  if (sb.language == 'federation standard') {
+    console.log(hello[this.language]);
+    return (hello['federation standard']);
+  }
+};
+//TODO: put this on the SentientBeing prototype
+
 // TODO: create three SentientBeings, one for each language in the
 // 'hello' object above.
-var klingon = new SentientBeing(); // TODO: make a klingon
-var romulan = new SentientBeing(); // TODO: make a romulan
-var human = new SentientBeing(); // TODO: make a human
+function Human() {}
+
+function Klingon() {}
 
-assert(human.sayHello(klingon) === 'nuqneH',
+function Romulan() {}
+
+var klingon = new SentientBeing("Qo\"Nos", "klingon");
+var romulan = new SentientBeing("Romulus", "romulan");
+var human = new SentientBeing("Earth", "federation standard");
+
+assert(Human.sayHello(Klingon) === 'nuqneH',
   'the klingon should hear nuqneH');
 
 // TODO: write five more assertions, to complete all the possible
 // greetings between the three types of sentient beings you created above.
+assert((new Human()).sayHello(new Romulan()) === 'Jolan\'tru',
```

Comment:
so it looks like you were going the direction of the advanced lab. Getting closer! For the beginning lab, it's okay to look at 
assert(human.sayHello(klingon) .....
if you're going to go the new Human route, then you need to create each species as a subclass of sentient beings (so they'll need sentientBeing as a prototype).